src/execute.rs

```diff
@@ -94,6 +96,55 @@ pub fn execute_mint(
         .add_attribute("color_map", format!("{:?}", new_color_map)))
 }
 
+pub fn execute_change_url(
+    deps: DepsMut,
+    info: MessageInfo,
+    env: Env,
+    position: u16,
+    url: String,
+) -> Result<Response, ContractError> {
+    let owner_check = get_owner(deps.as_ref(), env, position);
+    let owner;
+
+    match owner_check {
+        None => {
+            return Err(ContractError::DoesNotExist {});
+        }
+        Some(_) => owner = owner_check.unwrap(),
+    }
```

Comment:
nit: might want to make this a aux function because I imagine this check will be done multiple times in other places as well

Reply:
created a token_minted bool function for this